webhook/resources/volume/validator.go

```diff
@@ -341,6 +342,25 @@ func (v *volumeValidator) validateExpansionSize(oldVolume *longhorn.Volume, newV
 		return fmt.Errorf("shrinking volume %v size from %v to %v is not supported", newVolume.Name, oldSize, newSize)
 	}
 
+	replicas, err := v.ds.ListVolumeReplicasRO(newVolume.Name)
+	if err != nil {
+		return err
+	}
+
+	for _, replica := range replicas {
+		if datastore.IsBackendStoreDriverV1(replica.Spec.BackendStoreDriver) {
+			diskUUID := replica.Spec.DiskID
+			node, diskName, err := v.ds.GetReadyDiskNode(diskUUID)
+			if err != nil {
+				return err
+			}
+			diskStatus := node.Status.DiskStatus[diskName]
+			if (strings.HasPrefix(string(diskStatus.Type), "ext") && newSize >= util.MaxExt4VolumeSize) || (string(diskStatus.Type) == "xfs" && newSize >= util.MaxXfsVolumeSize) {
```

Comment:
```suggestion
			if (string(diskStatus.Type) == "ext4" && newSize >= util.MaxExt4VolumeSize) || (string(diskStatus.Type) == "xfs" && newSize >= util.MaxXfsVolumeSize) {
```

I think we should do an exact match check here. I'm honestly not sure whether it's possible to run Longhorn on top of ext3, but if it is, ext3 has a different maximum file size.

Reply:
Ah, I see your above comment that we cannot know which ext it is. So what do we see here? `ext2/ext3` as in my output below?

```
eweber@laptop:~/longhorn-manager> stat -f .
  File: "."
    ID: 23a4862bca551670 Namelen: 255     Type: ext2/ext3
Block size: 4096       Fundamental block size: 4096
Blocks: Total: 263940717  Free: 249947342  Available: 236521474
Inodes: Total: 67108864   Free: 66501058
```

If this is the case, I guess for now the `HasPrefix` makes sense. (Maybe we can put a similar comment here so it is clear why we do not do the exact match.

Reply:
>Ah, I see your above comment that we cannot know which ext it is. So what do we see here? ext2/ext3 as in my output below?

Yes, it is as in your output. The `stat -f` shows `ext2/ext3` for `ext4` file system.
Added the comment to the check statement.
Thanks.